test/e2e/json-rpc/eth_estimateGas.spec.js

```diff
@@ -0,0 +1,52 @@
+const { strict: assert } = require('assert');
+const { convertToHexValue, withFixtures } = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+
+describe('eth_estimateGas', function () {
+  const ganacheOptions = {
+    accounts: [
+      {
+        secretKey:
+          '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+        balance: convertToHexValue(25000000000000000000),
+      },
+    ],
+  };
+  it('executes a estimate gas json rpc call', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        // eth_estimateGas
+        await driver.openNewPage(`http://127.0.0.1:8080`);
+
+        const estimateGas = JSON.stringify({
+          jsonrpc: '2.0',
+          method: 'eth_estimateGas',
+          params: [
+            {
+              to: '0x5cfe73b6021e818b776b421b1c4db2474086a7e1',
+              from: '0x5cfe73b6021e818b776b421b1c4db2474086a7e1',
+            },
+          ],
+        });
```

Comment:
Q: why are we sending an RPC request for estimate gas if no transaction has been initiated? I wouldn't expect this RPC to actually happen, unless we triggered a send transaction to ourselves (`0x5cfe73b6021e818b776b421b1c4db2474086a7e1`) :thinking:

Reply:
oh I see, you are actually forcing the request, instead of waiting if it happened :)

Reply:
> This way we would test that these requests are correctly sent whenever they should and with the correct params.

I believe all the params for `eth_estimateGas` as optional. [Ethereum estimateGas](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_estimategas). There are talks in the works with the approach and implementation of fuzzing tests for the api to test the correct/incorrect params.

> Otherwise, it seems more to me that we are testing the API (which is outside of the testing scope and will change with each provider)

I thought this was the approach with the json rpc tests, a suite of tests that implements the json rpc methods?

Reply:
thank you @tmashuang I think I don't have the complete context around the goal of the json rpc tests. I've approved the Pr and trying to get more context on this separately